Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -332,8 +382,7 @@ FieldValue DecodeFieldValueImpl(pb_istream_t* stream) {
 void Writer::EncodeNestedMessage(
     const std::function<void(Writer*)>& encode_message_fn) {
   // First calculate the message size using a non-writing substream.
-  pb_ostream_t raw_sizing_substream = PB_OSTREAM_SIZING;
-  Writer sizing_substream(&raw_sizing_substream);
+  Writer sizing_substream = Writer::SizingStream();
```

Comment:
Optional: consider a shorter name than `sizing_substream`. Maybe just `sizer`? `writing_substream` could be `writer` too.